code/game/objects/items/devices/autopsy.dm

```diff
@@ -156,7 +157,8 @@
 	if(!can_operate(M))
 		return
 
-	if(target_name != M.name)
+	if(target_real_name != M.real_name)
```

Comment:
It is better to use `M.UID()` here if you want to track a mob itself. It will always be unique and will make it clear that you use it to track/reference something

Reply:
Huh, TIL about UID.
Done, thanks!